workflow/templateresolution/context.go

```diff
@@ -217,68 +225,65 @@ func (ctx *Context) resolveTemplateImpl(tmplHolder wfv1.TemplateHolder, depth in
 	// Update the template base of the context.
 	newTmplCtx, err := ctx.WithTemplateHolder(tmplHolder)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, false, err
 	}
 
 	// Return a concrete template without digging into it.
 	if tmpl.GetType() != wfv1.TemplateTypeUnknown {
-		return newTmplCtx, tmpl, nil
+		return newTmplCtx, tmpl, templateStored, nil
 	}
 
 	// Dig into nested references with new template base.
-	finalTmplCtx, resolvedTmpl, err := newTmplCtx.resolveTemplateImpl(tmpl, depth+1)
+	finalTmplCtx, resolvedTmpl, templateStoredInCall, err := newTmplCtx.resolveTemplateImpl(tmpl, depth+1)
```

Comment:
Since we don't resolve a template reference of a template other than `DagTask`, `WorkflowStep ` and `NodeStatus` anymore, we don't need to recursively resolve templates. What do you think?

Reply:
Oh this is a good insight! I will take a look 🙂